appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java

```diff
@@ -7,48 +7,96 @@
 import android.content.Intent;
 import android.content.pm.PackageManager;
 import android.net.Uri;
+import android.util.ArrayMap;
+import android.util.Log;
 import com.asf.appcoins.sdk.ads.BuildConfig;
 import com.asf.appcoins.sdk.ads.R;
+import java.lang.reflect.Field;
 
-public class WalletUtils{
+public class WalletUtils {
 
-  private static String walletPackageName = BuildConfig.BDS_WALLET_PACKAGE_NAME;
+  private static boolean dialogVisible;
 
-  public static boolean hasWalletInstalled(Context context) {
+  public static String walletPackageName = BuildConfig.BDS_WALLET_PACKAGE_NAME;
+
+  public static Context context;
+
+  public static boolean isDialogVisible() {
+    return dialogVisible;
+  }
+
+  public static void setContext(Context cont) {
+    context = cont;
```

Comment:
If this is a utility that provides static method, it should have have fields initialisation. If the idea is to have a class that manages the installation dialog and all the needed components to do so, please consider having a class that is initialises the field without them being static, and the method provided being non static.